[JSC] Move OMGOrigin to WasmOrigin and use it in BBQ as well

Constellation opened the pull request:
#### 08524c1b29b51e899d26b5eafa8d8d977f8321e8
<pre>
[JSC] Move OMGOrigin to WasmOrigin and use it in BBQ as well
<a href="https://bugs.webkit.org/show_bug.cgi?id=299257">https://bugs.webkit.org/show_bug.cgi?id=299257</a>
<a href="https://rdar.apple.com/161053703">rdar://161053703</a>

Reviewed by Justin Michaud.

This patch renames OMGOrigin to WasmOrigin, and uses it in BBQ as well.
This simplifies BBQ disassembler&apos;s code, and paves the way to get
CallSiteIndex from PC in BBQ as well as OMG. Also annotating late / slow
code with appropriate origins as well so sampling profiler can find them
correctly for BBQ.

* Source/JavaScriptCore/b3/B3Origin.h:
(JSC::B3::Origin::Origin):
(JSC::B3::Origin::isWasmOrigin const):
(JSC::B3::Origin::wasmOrigin const):
(JSC::B3::Origin::maybeWasmOrigin const):
(JSC::B3::Origin::isOMGOrigin const): Deleted.
(JSC::B3::Origin::omgOrigin const): Deleted.
(JSC::B3::Origin::maybeOMGOrigin const): Deleted.
* Source/JavaScriptCore/jit/PCToCodeOriginMap.h:
(JSC::Wasm::WasmOrigin::WasmOrigin):
(JSC::Wasm::OMGOrigin::OMGOrigin): Deleted.
* Source/JavaScriptCore/wasm/WasmBBQDisassembler.cpp:
(JSC::Wasm::BBQDisassembler::dumpVectorForInstructions):
(JSC::Wasm::BBQDisassembler::dumpForInstructions):
(JSC::Wasm::makeString): Deleted.
* Source/JavaScriptCore/wasm/WasmBBQDisassembler.h:
(JSC::Wasm::BBQDisassembler::setOpcode):
(JSC::Wasm::PrefixedOpcode::PrefixedOpcode): Deleted.
* Source/JavaScriptCore/wasm/WasmBBQJIT.cpp:
(JSC::Wasm::BBQJITImpl::BBQJIT::BBQJIT):
(JSC::Wasm::BBQJITImpl::BBQJIT::addLatePath):
(JSC::Wasm::BBQJITImpl::BBQJIT::emitEntryTierUpCheck):
(JSC::Wasm::BBQJITImpl::BBQJIT::addTopLevel):
(JSC::Wasm::BBQJITImpl::BBQJIT::emitLoopTierUpCheckAndOSREntryData):
(JSC::Wasm::BBQJITImpl::BBQJIT::endTopLevel):
(JSC::Wasm::BBQJITImpl::BBQJIT::origin):
(JSC::Wasm::BBQJITImpl::BBQJIT::willParseOpcode):
(JSC::Wasm::BBQJITImpl::BBQJIT::willParseExtendedOpcode):
(JSC::Wasm::parseAndCompileBBQ):
* Source/JavaScriptCore/wasm/WasmBBQJIT.h:
* Source/JavaScriptCore/wasm/WasmBBQJIT64.cpp:
(JSC::Wasm::BBQJITImpl::BBQJIT::emitAllocateGCArrayUninitialized):
(JSC::Wasm::BBQJITImpl::BBQJIT::emitAllocateGCStructUninitialized):
* Source/JavaScriptCore/wasm/WasmCallee.cpp:
(JSC::Wasm::OptimizingJITCallee::materializePCToOriginMap):
* Source/JavaScriptCore/wasm/WasmCompilationContext.h:
* Source/JavaScriptCore/wasm/WasmOMGIRGenerator.cpp:
(JSC::Wasm::OMGIRGenerator::addThrow):
(JSC::Wasm::OMGIRGenerator::addThrowRef):
(JSC::Wasm::OMGIRGenerator::addRethrow):
(JSC::Wasm::OMGIRGenerator::origin):
(JSC::Wasm::parseAndCompileOMG):
* Source/JavaScriptCore/wasm/WasmOMGIRGenerator32_64.cpp:
(JSC::Wasm::OMGIRGenerator::addThrow):
(JSC::Wasm::OMGIRGenerator::addThrowRef):
(JSC::Wasm::OMGIRGenerator::addRethrow):
(JSC::Wasm::OMGIRGenerator::origin):
(JSC::Wasm::parseAndCompileOMG):
* Source/JavaScriptCore/wasm/WasmOpcodeOrigin.cpp:
(JSC::Wasm::OpcodeOrigin::opcodeString const):
(JSC::Wasm::OpcodeOrigin::dump const):
* Source/JavaScriptCore/wasm/WasmOpcodeOrigin.h:

Canonical link: <a href="https://commits.webkit.org/300303@main">https://commits.webkit.org/300303@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/407643568389849cf404b698af69d8ba8510c87c

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/122072 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/41775 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/32444 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/128636 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/74166 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/5f296179-6889-4942-b792-63514a08ff76) 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/42489 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/50368 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/92785 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/61668 "Passed tests") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/d7cc7928-2f6d-4b85-8559-abdd280b8d68) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/125024 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/33885 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/109317 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/73441 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/35ed67e6-269f-4bd9-9c52-8e145c2b9204) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/32898 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/27483 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/72130 "Built successfully") | | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/114220 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/103392 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/27674 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/131396 "Built successfully") | | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/120598 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/49011 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/37277 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/101347 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/49385 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/105531 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/101218 "Passed tests") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/46582 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/24701 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/45748 "Built successfully") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/19312 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/48868 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/54602 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/150756 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/48338 "Built successfully") | | [❌ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/38576 "Found 1 new JSC stress test failure: stress/dont-link-virtual-calls-on-compiler-thread.js.no-llint (failure)") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/51688 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/50018 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/407643568389849cf404b698af69d8ba8510c87c. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/51084

Constellation committed:
[JSC] Move OMGOrigin to WasmOrigin and use it in BBQ as well
https://bugs.webkit.org/show_bug.cgi?id=299257
rdar://161053703

Reviewed by Justin Michaud.

This patch renames OMGOrigin to WasmOrigin, and uses it in BBQ as well.
This simplifies BBQ disassembler's code, and paves the way to get
CallSiteIndex from PC in BBQ as well as OMG. Also annotating late / slow
code with appropriate origins as well so sampling profiler can find them
correctly for BBQ.

* Source/JavaScriptCore/b3/B3Origin.h:
(JSC::B3::Origin::Origin):
(JSC::B3::Origin::isWasmOrigin const):
(JSC::B3::Origin::wasmOrigin const):
(JSC::B3::Origin::maybeWasmOrigin const):
(JSC::B3::Origin::isOMGOrigin const): Deleted.
(JSC::B3::Origin::omgOrigin const): Deleted.
(JSC::B3::Origin::maybeOMGOrigin const): Deleted.
* Source/JavaScriptCore/jit/PCToCodeOriginMap.h:
(JSC::Wasm::WasmOrigin::WasmOrigin):
(JSC::Wasm::OMGOrigin::OMGOrigin): Deleted.
* Source/JavaScriptCore/wasm/WasmBBQDisassembler.cpp:
(JSC::Wasm::BBQDisassembler::dumpVectorForInstructions):
(JSC::Wasm::BBQDisassembler::dumpForInstructions):
(JSC::Wasm::makeString): Deleted.
* Source/JavaScriptCore/wasm/WasmBBQDisassembler.h:
(JSC::Wasm::BBQDisassembler::setOpcode):
(JSC::Wasm::PrefixedOpcode::PrefixedOpcode): Deleted.
* Source/JavaScriptCore/wasm/WasmBBQJIT.cpp:
(JSC::Wasm::BBQJITImpl::BBQJIT::BBQJIT):
(JSC::Wasm::BBQJITImpl::BBQJIT::addLatePath):
(JSC::Wasm::BBQJITImpl::BBQJIT::emitEntryTierUpCheck):
(JSC::Wasm::BBQJITImpl::BBQJIT::addTopLevel):
(JSC::Wasm::BBQJITImpl::BBQJIT::emitLoopTierUpCheckAndOSREntryData):
(JSC::Wasm::BBQJITImpl::BBQJIT::endTopLevel):
(JSC::Wasm::BBQJITImpl::BBQJIT::origin):
(JSC::Wasm::BBQJITImpl::BBQJIT::willParseOpcode):
(JSC::Wasm::BBQJITImpl::BBQJIT::willParseExtendedOpcode):
(JSC::Wasm::parseAndCompileBBQ):
* Source/JavaScriptCore/wasm/WasmBBQJIT.h:
* Source/JavaScriptCore/wasm/WasmBBQJIT64.cpp:
(JSC::Wasm::BBQJITImpl::BBQJIT::emitAllocateGCArrayUninitialized):
(JSC::Wasm::BBQJITImpl::BBQJIT::emitAllocateGCStructUninitialized):
* Source/JavaScriptCore/wasm/WasmCallee.cpp:
(JSC::Wasm::OptimizingJITCallee::materializePCToOriginMap):
* Source/JavaScriptCore/wasm/WasmCompilationContext.h:
* Source/JavaScriptCore/wasm/WasmOMGIRGenerator.cpp:
(JSC::Wasm::OMGIRGenerator::addThrow):
(JSC::Wasm::OMGIRGenerator::addThrowRef):
(JSC::Wasm::OMGIRGenerator::addRethrow):
(JSC::Wasm::OMGIRGenerator::origin):
(JSC::Wasm::parseAndCompileOMG):
* Source/JavaScriptCore/wasm/WasmOMGIRGenerator32_64.cpp:
(JSC::Wasm::OMGIRGenerator::addThrow):
(JSC::Wasm::OMGIRGenerator::addThrowRef):
(JSC::Wasm::OMGIRGenerator::addRethrow):
(JSC::Wasm::OMGIRGenerator::origin):
(JSC::Wasm::parseAndCompileOMG):
* Source/JavaScriptCore/wasm/WasmOpcodeOrigin.cpp:
(JSC::Wasm::OpcodeOrigin::opcodeString const):
(JSC::Wasm::OpcodeOrigin::dump const):
* Source/JavaScriptCore/wasm/WasmOpcodeOrigin.h:

Canonical link: https://commits.webkit.org/300303@main
